psydac/linalg/tests/test_block.py

```diff
@@ -52,23 +53,140 @@ def test_block_linear_operator_serial_dot( n1, n2, p1, p2, P1, P2  ):
     cart = CartDecomposition(D, npts, global_starts, global_ends, pads=[p1,p2], shifts=[1,1])
 
     # Create vector spaces, stencil matrices, and stencil vectors
-    V = StencilVectorSpace( cart )
-    M1 = StencilMatrix( V, V )
+    V = StencilVectorSpace( cart, dtype=dtype )
+    M1 = StencilMatrix( V, V)
     M2 = StencilMatrix( V, V )
     M3 = StencilMatrix( V, V )
     x1 = StencilVector( V )
     x2 = StencilVector( V )
 
     # Fill in stencil matrices based on diagonal index
+    if dtype==complex:
+        f=lambda k1,k2: 10j*k1+k2
+    else:
+        f=lambda k1,k2: 10*k1+k2
+
     for k1 in range(-p1,p1+1):
         for k2 in range(-p2,p2+1):
-            M1[:,:,k1,k2] = 10*k1+k2
-            M2[:,:,k1,k2] = 10*k1+k2+2.
-            M3[:,:,k1,k2] = 10*k1+k2+5.
+            M1[:,:,k1,k2] = f(k1,k2)
+            M2[:,:,k1,k2] = f(k1,k2)+2.
+            M3[:,:,k1,k2] = f(k1,k2)+5.
+
     M1.remove_spurious_entries()
     M2.remove_spurious_entries()
     M3.remove_spurious_entries()
 
+    # Fill in vector with random values, then update ghost regions
+    for i1 in range(n1):
+        for i2 in range(n2):
+            x1[i1,i2] = 2.0*random() - 1.0
+            x2[i1,i2] = 5.0*random() - 1.0
+    x1.update_ghost_regions()
+    x2.update_ghost_regions()
```

Comment:
If this function only tests the space, why are you creating vectors and matrices? The code I highlighted may be removed.

Reply:
[86b06e9](https://github.com/pyccel/psydac/pull/273/commits/86b06e97fda7a028f2d6ec3f23b6c81d44eaf696)